### ISSUE TITLE
proxyForInterface should be in twisted.python.components, not twisted.python.util

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2928]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#2928|
|Type|enhancement|
|Created|2007-11-28 17:47:26Z|
|Branch|https://github.com/twisted/twisted/tree/move-proxyForInterface-2928|

"components" is an appropriate module for dealing with zope.interface abstractions.  "util" isn't really an appropriate module for dealing with anything.

<details><summary>Searchable metadata</summary>

```
trac-id__2928 2928
type__enhancement enhancement
reporter__glyph glyph
priority__highest highest
milestone__ 
branch__branches_move_proxyForInterface_2928 branches/move-proxyForInterface-2928
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1196272046000000 1196272046000000
changetime__1196554234000000 1196554234000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml commented|
|-|-|

(In [22000]) Branching to 'move-proxyForInterface-2928'

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml commented|
|-|-|

This looked so easy I thought I'd do it. (2 minute rule etc). You can review it :)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @jml<br>@glyph set status to `new`|
|-|-|

`test-case-name` on both modules is now wrong.  +1 on everything else though, including moving the test modules.  please update those and merge.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_6"></a>|@jml set status to `closed`|
|-|-|

(In [22015]) Move proxyForInterface to t.python.components.

 * Author: jml
 * Reviewer: glyph
 * Fixes [#2928](https://github.com/twisted/twisted/issues/2928)

Also moves the tests for python.util and python.components into 
twisted.python.test.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
